src/pipe.test.ts

```diff
@@ -115,4 +117,39 @@ describe("lazy", () => {
     expect(count2).toHaveBeenCalledTimes(2);
     expect(result).toEqual([100, 200]);
   });
+
+  it("lazy take + flat", () => {
```

Comment:
Can you rename the test to reflect the issue better, take and flat are just a way to reproduce it. I'd call this "lazy early exit with hasMany"

Reply:
Renamed it according to your suggestion